### PR TITLE
Fix update-ips fetch swift url

### PR DESCRIPTION
**Pull request by robinAwallace**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [ ] personal data beyond what is necessary for interacting with this pull request, nor
> - [ ] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [x] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

I noticed that the update-ips did not find the swift url. 
Im unsure if it is my version of curl that returns 1 less line. 
If that is the case, we probably need to find a more stable way extract the url from the response.

```
curl --version
curl 7.88.1 (x86_64-pc-linux-gnu) libcurl/7.88.1 OpenSSL/3.0.8 zlib/1.2.13 brotli/1.0.9 zstd/1.5.4 libidn2/2.3.3 libpsl/0.21.2 (+libidn2/2.3.3) libssh/0.10.4/openssl/zlib nghttp2/1.52.0 librtmp/2.3
Release-Date: 2023-02-20
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

```

#### Checklist

- [ ] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Documentation checks:
  - [ ] The [public documentation](https://github.com/elastisys/compliantkubernetes) required no updates
  - [ ] The [public documentation](https://github.com/elastisys/compliantkubernetes) required an update - [link to change](set-me\)
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: bin/update-ips.bash [hunk 1]
```diff
@@ -223,7 +223,7 @@ get_swift_url() {
   fi
 
   os_token=$(echo "${response}" | grep -oP "x-subject-token:\s+\K\S+")
-  swift_url=$(echo "${response}" | tail -n +15 | jq -r '.[].catalog[] | select( .type == "object-store" and .name == "swift") | .endpoints[] | select(.interface == "public" and .region == "'"${swift_region}"'") | .url')
+  swift_url=$(echo "${response}" | tail -n +14 | jq -r '.[].catalog[] | select( .type == "object-store" and .name == "swift") | .endpoints[] | select(.interface == "public" and .region == "'"${swift_region}"'") | .url')
```
Comment (aarnq) [inline bin/update-ips.bash:226]: Is there any way to make this not rely on specific line numbers?

Comment (robinAwallace) [reply bin/update-ips.bash:226]: I can change it to calls to the api, then i can skip the complete tail part. But I can test around a bit more

File: REQUIREMENTS [hunk 2]
```diff
@@ -1,5 +1,5 @@
 pkg:deb/ubuntu/apache2-utils@2.4.52*?arch=amd64&distro=jammy
-pkg:deb/ubuntu/curl@7.81.0*?arch=amd64&distro=jammy
+pkg:deb/ubuntu/curl@7.83.0*?arch=amd64&distro=jammy
```
Comment (simonklb) [inline REQUIREMENTS:2]: curl 7.83.0 is not available in jammy

**Review by simonklb — CHANGES_REQUESTED** [review]

File: bin/update-ips.bash [hunk 3]
```diff
@@ -182,8 +182,11 @@ get_swift_url() {
   auth_url="$(yq_read "sc" "${swift_config_option}.authUrl" "")"
   swift_region="$(yq_read "sc" "${swift_config_option}.region" "")"
 
+  headers=$(mktemp --suffix="_headers")
+  append_trap "rm ${headers}" EXIT
+
   if [ -n "$(yq_read_secret "${swift_config_option}.username" "")" ]; then
-    response=$(curl -i -s -H "Content-Type: application/json" -d '
+    response=$(curl -s -D "${headers}" -H "Content-Type: application/json" -d '
```
[the remark this answers is not in the document]
Comment (simonklb) [reply bin/update-ips.bash:189]: (or just `-i` instead of `-D /dev/stdout` like we did before :laughing:)

**Review by lunkan93 — APPROVED** [review]
I like the idea of looking at @simonklb suggestions in a new PR so that we can include this in the v0.41 release.

**Pull request merged**